cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -339,28 +313,28 @@ struct column_row_comparator_fn {
       }
     }
 
-    return compare<Type>(lhs.element<Type>(i), lhs.element<Type>(j));
+    return compare(lhs.element<Type>(i), lhs.element<Type>(j));
   }
 };
 
 /**
  * @brief Struct used in type_dispatcher for comparing two entries in a lists column.
  */
 struct column_row_comparator_dispatch {
-  offset_type const* const list_offsets;
+  size_type const* const list_indices;
```

Comment:
This looks to be redundant with `element_equality_comparator`. Likewise, the `table_row_comparator_fn` looks to be redundant with `row_equality_comparator`.

Reply:
`column_row_comparator_dispatch` is not the same as `element_equality_comparator`. In particular, it allows to have different output depending on the `nans_equal` parameter. `element_equality_comparator` does not offer such feature.
Sorry that I didn't clarify it in @brief doxygen (now added).

Reply:
That's a relatively minor thing to duplicate such a large bit of complicated code. Remind me why `nans_equal` is necessary? For the most part, we've resigned to just always treat NaNs as equal in most libcudf functions.

Reply:
We had a discussion before regarding that with Keith and decided to have an option allowing 2 different behaviors because:
1. Pandas wants to have NaNs treated as equal
2. Spark wants to have NaNs considered unequal

Reply:
> Spark wants to have NaNs considered unequal

Seriously? Most of the code in libcudf that's specialized to treat NaNs as equal was because of Spark. How wonderfully inconsistent.

Reply:
However, Spark is "consistently inconsistent" thus in groupby it treats NaNs as equal for NaN keys (but not values). That's why we didn't need to modify groupby code for floating-point keys.

Reply:
For future work I'd like to see adding a static/dynamic option to the row/element comparator for controlling NaN equality so we don't have to maintain this code that is 99% redundant.